app/grandchallenge/core/templates/grandchallenge/partials/userlinks.html

```diff
@@ -42,19 +42,31 @@ <h6 class="dropdown-header">{{ request.site.name }} Support</h6>
     </div>
 </li>
 {% if user.is_authenticated %}
-    {% if ACTSTREAM_ENABLE %}
     <li class="nav-item mr-auto">
-        {% if user.user_profile.has_unread_notifications %}
+        {% if user.directmessageunreadby_set.exists %}
```

Comment:
Since this feature is only available to challenge admins (and when messaged to participants), shouldn't we hide the entire icon if there are no messages?

Reply:
Yes, thought about that but then it would be an extra lookup on every single page. We don't have an abstract user so I couldn't easily work out how to attach a `prefetch_related` to `request.user`. So, knowing that we have to make one extra call for this I went for the quickest one that also avoids any joins.

In the longer term we could move both the messages and notifications icon and alert to a HTMX view instead, then we don't slow down each page load.

Reply:
Fair enough! I hadn't thought this far yet, but I agree that it's definitely not worth the extra call and also not really worth adding a custom manager for the user model to do the necessary prefetching.